File: 0415.字符串相加/0415-字符串相加.java
class Solution {
    public String addStrings(String num1, String num2) {
        StringBuffer ans = new StringBuffer();
        int i = num1.length()-1;
        int j = num2.length()-1;
        int val = 0;
        int val1 = 0;
        int val2 = 0;

        while( i>=0 || j>=0 || val>0 ) {
            val1 = i>=0 ? num1.charAt(i)-'0' : 0 ;
            val2 = j>=0 ? num2.charAt(j)-'0' : 0;
            i--;
            j--;

            val += val1 + val2;
            ans.append(val % 10);
            val = val/10;
        }

        ans.reverse();
        return ans.toString();
    }
}

File: 0415.字符串相加/0415-字符串相加.py
class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        ans = ""
        i = len(num1)-1
        j = len(num2)-1
        val = 0

        while i>=0 or j>=0 or val>0:
            val1 = int(num1[i]) if i>=0 else 0
            val2 = int(num2[j]) if j>=0 else 0
            i, j = i-1, j-1

            val = val + val1 + val2;
            ans = str(val%10) + ans
            val = val//10

        return ans